src/http/modules/ngx_http_ssi_filter_module.c

```diff
@@ -2232,12 +2232,30 @@ ngx_http_ssi_include(ngx_http_request_t *r, ngx_http_ssi_ctx_t *ctx,
 static ngx_int_t
 ngx_http_ssi_stub_output(ngx_http_request_t *r, void *data, ngx_int_t rc)
 {
-    ngx_chain_t  *out;
+    ngx_chain_t        *out;
+    ngx_http_ssi_ctx_t *ctx;
 
     if (rc == NGX_ERROR || r->connection->error || r->request_output) {
         return rc;
     }
 
+    ctx = ngx_http_get_module_ctx(r, ngx_http_ssi_filter_module);
+
+    if (ctx && ctx->stub_output_done) {
+        return NGX_OK;
+    }
+
+    if (ctx == NULL) {
+        ctx = ngx_pcalloc(r->pool, sizeof(ngx_http_ssi_ctx_t));
+        if (ctx == NULL) {
+            return NGX_ERROR;
+        }
+
+        ngx_http_set_ctx(r, ctx, ngx_http_ssi_filter_module);
```

Comment:
As far as I can tell, setting context here causes Nginx to segfault, but I don't know why or how to debug it from the prove testing script. I'm at a loss at the moment.